portfolio/src/main/client/src/util/html.ts

```diff
@@ -1,10 +1,19 @@
 import type {Arrunk} from './types';
 
 /**
- * Any item that can be rendered by interpolating it with the @see htmlElement
- * function.
+ * Any stateless item that can be rendered by interpolating it with the @see
+ * htmlElement function.
  */
-export type Renderable = Arrunk<string | Node | null | undefined>;
+export type PrimitiveRenderable = Arrunk<string | Node | null | undefined>;
+
+/**
+ * A component renderable, which has a certain state and renders a primitive.
+ */
+export interface ComponentRenderable {
+  render(): string | Node | null;
+}
```

Comment:
In which situations would you need this Renderable interface? I imagine that in a follow up PR, you'll have code like:
```
const auth = Authentication();
return htmls`Foo bar ${auth} baz`;
```

Why not call render (or better yet just provide a `root` variable so it turns into:
```
const auth = Authentication();
return htmls`Foo bar ${auth.root} baz`;
```

Reply:
Basically, I think that hiding the calls to `render` to be further down the stack makes this code harder to read and predict.